### PR TITLE
Complete repository tests

### PR DESCRIPTION
**Pull request by kaizimmerm**
- My original intend with this pull requests was to close a few holes in our repository tests. Especially Target search/filter tests needed some love.
- However, doing that I figured that our equals/hashcode policy (primary key, opt lock revision) is not in place anymore for many entities. I fixed that.
- As always when I move through the repository I found lots of boilerplate code including unnecessary JavaDoc. I found also some errors in the docs as well.

Note: this pull request is based on [https://github.com/eclipse/hawkbit/pull/104] . So I recommend merging that one first to have a better picture what comes with this one.

### CODE REVIEW

File: hawkbit-repository/src/main/java/org/eclipse/hawkbit/repository/ControllerManagement.java [hunk 1]
```diff
@@ -376,16 +376,10 @@ public Action handleAddUpdateActionStatus(final ActionStatus actionStatus, final
...
-            mergedAction.setStatus(Status.ERROR);
-            mergedTarget.setAssignedDistributionSet(null);
-            targetManagement.updateTarget(mergedTarget);
+            handleErrorOnAction(mergedAction, mergedTarget);
             break;
         case FINISHED:
-            // set action inactive
-            mergedAction.setActive(false);
-            mergedAction.setStatus(Status.FINISHED);
+
```
Comment (michahirsch) [inline hawkbit-repository/src/main/java/org/eclipse/hawkbit/repository/ControllerManagement.java]: remove this empty line

File: hawkbit-repository/src/main/java/org/eclipse/hawkbit/repository/ControllerManagement.java [hunk 2]
```diff
@@ -404,6 +398,14 @@ public Action handleAddUpdateActionStatus(final ActionStatus actionStatus, final
         return actionRepository.save(mergedAction);
     }
 
+    private void handleErrorOnAction(final Action mergedAction, final Target mergedTarget) {
+        // set action inactive
```
Comment (SirWayne) [inline hawkbit-repository/src/main/java/org/eclipse/hawkbit/repository/ControllerManagement.java]: useless comment

File: hawkbit-repository/src/main/java/org/eclipse/hawkbit/repository/model/SoftwareModuleMetadata.java [hunk 3]
```diff
@@ -69,56 +65,83 @@ public SoftwareModuleMetadata(final String key, final SoftwareModule softwareMod
...
+            return false;
+        }
+        if (softwareModule == null) {
+            if (other.softwareModule != null) {
+                return false;
+            }
+        } else if (!softwareModule.equals(other.softwareModule)) {
+            return false;
+        }
+        if (value == null) {
```
Comment (SirWayne) [inline hawkbit-repository/src/main/java/org/eclipse/hawkbit/repository/model/SoftwareModuleMetadata.java]: Do not check the value

File: hawkbit-repository/src/main/java/org/eclipse/hawkbit/repository/model/SoftwareModuleMetadata.java [hunk 4]
```diff
@@ -69,56 +65,83 @@ public SoftwareModuleMetadata(final String key, final SoftwareModule softwareMod
...
+
+    @Override
+    public boolean equals(final Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
```
Comment (SirWayne) [inline hawkbit-repository/src/main/java/org/eclipse/hawkbit/repository/model/SoftwareModuleMetadata.java]: getClass or instance of?

File: hawkbit-repository/src/main/java/org/eclipse/hawkbit/repository/model/TargetInfo.java [hunk 5]
```diff
@@ -386,15 +309,41 @@ public LocalDateTime getCurrentDate() {
...
+            return false;
+        }
+        final TargetInfo other = (TargetInfo) obj;
+        if (target == null) {
+            if (other.target != null) {
+                return false;
+            }
+        } else if (!target.equals(other.target)) {
+            return false;
+        }
```
Comment (SirWayne) [inline hawkbit-repository/src/main/java/org/eclipse/hawkbit/repository/model/TargetInfo.java]: Check primary key (targetId) too

**Comment by SirWayne** [comment]
:+1:

**Pull request merged**